internal/vfs/s3fs.go

```diff
@@ -159,8 +159,8 @@ func (fs *S3Fs) Stat(name string) (os.FileInfo, error) {
 	}
 	obj, err := fs.headObject(name)
 	if err == nil {
-		// a "dir" has a trailing "/" so we cannot have a directory here
-		return updateFileInfoModTime(fs.getStorageID(), name, NewFileInfo(name, false, obj.ContentLength,
+		isDir := *obj.ContentType == s3DirMimeType
```

Comment:
Please use `util.GetStringFromPointer(obj.ContentType)`

Reply:
also add a comment explaining why we may not have the trailing slash and so we check the mime type

Reply:
> Please use `util.GetStringFromPointer(obj.ContentType)`

It's so important to use util, because it can throw:
`invalid memory address or nil pointer dereference` error on FTP put or FTP size.

`{"level":"warn","time":"2022-11-15T13:41:39.737","sender":"ftpserverlib","server_id":"FTP_0","clientId":"8","err":"runtime error: invalid memory address or nil pointer dereference","command":"SIZE","param":"/some-file","message":"Internal command handling error"}`

Reply:
I did as you requested.